packages/frontend/src/components/DebouncedSearchInput/index.tsx

```diff
@@ -13,31 +18,54 @@ export default function DebouncedSearchInput({
   onChange,
   searchValue,
 }: DebouncedSearchInputProps): React.ReactElement {
+  const [tempSearchValue, setTempSearchValue] = useState(searchValue)
+
   const handleSearchInputChangeDebounced = useMemo(
     () => debounce(onChange, 500),
     [onChange],
   )
 
   const onSearchInputChange = useCallback(
     (event: React.ChangeEvent<HTMLInputElement>) => {
+      setTempSearchValue(event.target.value)
       handleSearchInputChangeDebounced(event.target.value)
     },
     [handleSearchInputChangeDebounced],
   )
 
+  const clearSearch = useCallback(() => {
+    setTempSearchValue('')
+    onChange('')
+    handleSearchInputChangeDebounced.cancel()
+  }, [onChange, handleSearchInputChangeDebounced])
+
   return (
     <InputGroup>
       <InputLeftElement h="100%">
-        <Icon as={BiSearch} style={{ height: 20, width: 20 }} />
+        <Icon as={BiSearch} h={5} w={5} />
       </InputLeftElement>
       <Input
         type="text"
-        w="full"
+        w={{ base: '100%', md: '20rem' }}
+        maxW="100%"
```

Comment:
```suggestion
        w={{ base: '100%', md: '21rem' }}
        maxW="100%" // feels like this is not needed?
```
20rem looks like the icon is positioned too much to the right
<img width="1201" alt="image" src="https://github.com/user-attachments/assets/de62eb04-1e3a-40c3-97d5-6bac79228e52">